client/app/components/ParameterValueInput.jsx

```diff
@@ -34,108 +32,108 @@ export class ParameterValueInput extends React.Component {
     enumOptions: '',
     queryId: null,
     parameter: null,
-    applyButton: false,
     onSelect: () => {},
     className: '',
   };
 
   constructor(props) {
     super(props);
-    this.state = { value: props.value };
+    this.state = {
+      value: props.value,
+      isDirty: false,
+    };
   }
 
-  renderApplyButton() {
-    const { onSelect } = this.props;
-    const { value } = this.state;
-    return (
-      <Button
-        className="parameter-apply-button"
-        type="primary"
-        size="small"
-        onClick={() => onSelect(value)}
-      >
-        Apply
-      </Button>
-    );
+  componentDidUpdate = (prevProps) => {
+    // if value prop updated, reset dirty state
+    if (prevProps.value !== this.props.value) {
+      this.setState({ isDirty: false });
```

Comment:
> I wonder if the state value should also be updated here. It doesn't seem a good React practice, but there are some components that use that state as value and not the prop.

My code is assuming that value changes can only come from direct user interaction, so in that case there won't be prop/state syncing issues. But in case the prop is changed from outside (for instance, another user changed the param and it auto-updated) this would cause such an issue. I just thought it would be highly unlikely so I left it like this. I can add it just to be sure.

> A second thought: Is the value state really necessary now? (it was for the individual Apply button)

I think since the "apply button" changes the `prop.value` indirectly and not the actual component value update, we have to use `state.value`. Or do you mean, using `defaultValue={this.props.value}` instead?

Reply:
> My code is assuming that value changes can only come from direct user interaction, so in that case there won't be prop/state syncing issues.

:+1: 

> Or do you mean, using defaultValue={this.props.value} instead?

Exactly that.

Reply:
BUT I want to leave it open for an "undo" feature, therefore would need state. Wdyt?

Reply:
> BUT I want to leave it open for an "undo" feature, therefore would need state. Wdyt?

Yeah, makes sense then :grin:

Reply:
Added `value` to the state update in `componentDidUpdate`